File: src/main/java/com/learnershi/rclasssocket/entity/Badge.java
package com.learnershi.rclasssocket.entity;


import com.learnershi.rclasssocket.entity.enums.BadgeType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * 뱃지
 */
@Accessors(chain = true)
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Document
public class Badge {

    private String id;

    // 교실 id
    private String classRoomId;

    // 유저 seq
    private String userSeq;

    // 뱃지 타입
    private BadgeType badgeType;

    public String getBadgeType() {
        return badgeType.getValue();
    }

}

File: src/main/java/com/learnershi/rclasssocket/entity/Badge.kt
package com.learnershi.rclasssocket.entity

import com.learnershi.rclasssocket.entity.enums.BadgeType
import org.springframework.data.mongodb.core.mapping.Document

/**
 * 뱃지
 */
@Document
class Badge {
    var id: String? = null

    // 교실 id
    var classRoomId: String? = null

    // 유저 seq
    var userSeq: String? = null

    // 뱃지 타입
    var badgeType: BadgeType = BadgeType.NONE
}